andes/routines/tds.py

```diff
@@ -365,6 +350,14 @@ def run(self, no_summary=False, from_csv=None, **kwargs):
         else:  # resume simulation
             self.init_resume()
 
+        # load from csv is provided
+        if from_csv is not None:
+            self.from_csv = from_csv
+        else:
+            self.from_csv = system.options.get("from_csv")
+        if self.from_csv is not None:
+            self.data_csv = self._load_csv(self.from_csv)
+
```

Comment:
``_load_csv()`` after ``init()``, to ensure smooth init when from_csv is given in ``TDS.run()``

Reply:
The issue is when CSV is given at "andes.load(from_csv='xx.csv')", ``TDS.init()`` run into error for the incompatible data size. It is because ``_load_csv()`` is called after ``TDS.init()`` in ``TDS.run()``.

Reply:
We can forget this as the introduced helper function ``TDS._csv_output()`` addressed it.